### PR TITLE
Release 0.4.0 Dark mode!

### PR DESCRIPTION
**Pull request by PatelUtkarsh**
## Version 0.4.0 Dark mode!

For the full list of changes in this release, please see the [Milestone 0.4.0](https://github.com/material-components/material-design-for-wordpress/issues?q=is%3Aissue+milestone%3A0.4.0).

https://user-images.githubusercontent.com/5015489/132191348-a06e9385-e6d2-431a-b5b4-7f263b741d7d.mp4

## Changelog:
-  **Dark mode**:  Pick a dark mode from an existing theme or create your own custom theme and generate a customized dark theme from it.
    - Manually turn on/off dark mode or automatically based on the user's system preference. 
    - Allow user to switch between light and dark mode via switcher.
- Add inline Icon in paragraphs. (https://github.com/material-components/material-design-for-wordpress/issues/79)
- Add meta tag in header (https://github.com/material-components/material-design-for-wordpress/issues/192)
- Fix breaking material list block in WordPress 5.8 (https://github.com/material-components/material-design-for-wordpress/issues/187)
- Color utility conversion (https://github.com/material-components/material-design-for-wordpress/issues/168)
- Add newsletter email signup (https://github.com/material-components/material-design-for-wordpress/issues/161)
- Add warning for bumping the minimum supported version to WordPress 5.6. (https://github.com/material-components/material-design-for-wordpress/issues/179)
## Props
Thanks to the many contributors who made this release possible through work on development, design, testing, project management, and more:

> @emeaguiar, @pbearne, @csossi, @jauyong, @mattchungxwp, @dawidmlynarz, @rodydavis, @PatelUtkarsh

### CODE REVIEW

**Comment by pbearne** [comment]
@googlebot I consent

**Comment by emeaguiar** [comment]
@googlebot I consent.

**Comment by rodydavis** [comment]
Let me know when this is ready to review!

File: plugin/assets/css/src/admin.css [hunk 1]
```diff
@@ -84,3 +84,23 @@
...
+
+.material_design_update_message {
+	background-color: #f5f5f5;
+	padding: 1rem;
+	font-weight: bold;
+	display: block;
+	margin-top: 1rem;
+	border: 1px solid #ccd0d4;
+
+	&::before {
```
Comment (rodydavis) [inline plugin/assets/css/src/admin.css:96]: This is sass and not css right?

File: plugin/assets/css/src/components/material.css [hunk 2]
```diff
@@ -29,6 +29,11 @@
 	direction: ltr;
 	font-feature-settings: "liga";
 	-webkit-font-smoothing: antialiased;
+
+	@nest p & {
```
Comment (rodydavis) [inline plugin/assets/css/src/components/material.css:33]: is there a fall back for where nesting is not supported?

File: plugin/assets/css/src/customize-controls.css [hunk 3]
```diff
@@ -59,18 +59,35 @@
 .customize-control-image-radio-wrap {
 	display: flex;
 	flex-wrap: wrap;
+
+	& h3 {
```
Comment (rodydavis) [inline plugin/assets/css/src/customize-controls.css:63]: Nested css fallback?

File: plugin/assets/css/src/customize-controls.css [hunk 4]
```diff
@@ -59,18 +59,35 @@
...
+	flex: 0 100%;
+	margin-bottom: 1px;
+	padding: 0 15px;
 
 	& img {
-		margin-top: 4px;
-		width: 100%;
+		border-radius: 5px;
+		height: 40px;
+		margin-bottom: 10px;
```
Comment (rodydavis) [inline plugin/assets/css/src/customize-controls.css:78]: Nested css fallback?

File: plugin/assets/css/src/gsm/content.css [hunk 5]
```diff
@@ -25,4 +25,9 @@
 		text-align: left;
 		width: 105px;
 	}
+
+	& .mdc-button--offsite .mdc-button__icon {
```
Comment (rodydavis) [inline plugin/assets/css/src/gsm/content.css:29]: Nested css fallback?

File: plugin/assets/css/src/wizard/content.css [hunk 6]
```diff
@@ -41,4 +41,9 @@
 	& .material-wizard__card img {
 		width: 100%;
 	}
+
+	& .mdc-button--offsite .mdc-button__icon {
```
Comment (rodydavis) [inline plugin/assets/css/src/wizard/content.css:45]: is this valid css?

File: plugin/assets/src/customizer/components/color-control/style.css [hunk 7]
```diff
@@ -0,0 +1,27 @@
+.material-design-color__picker {
+	align-items: center;
+	display: flex;
+	margin-bottom: 15px;
+
+	& .components-base-control .components-base-control__field {
```
Comment (rodydavis) [inline plugin/assets/src/customizer/components/color-control/style.css:6]: Is there a fallback for browsers that dont support nested css?

File: plugin/assets/src/customizer/components/style-settings-control/style.css [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+.style-settings-group {
+
+	& .style-settings-group__title {
```
Comment (rodydavis) [inline plugin/assets/src/customizer/components/style-settings-control/style.css:19]: is there a fallback?

File: plugin/tests/e2e/config/bootstrap.js [hunk 9]
```diff
@@ -205,8 +205,8 @@ beforeAll( async () => {
 		width: 1680,
 		height: 948,
 	} );
-	await page.setDefaultNavigationTimeout( 10000 );
-	await page.setDefaultTimeout( 10000 );
+	await page.setDefaultNavigationTimeout( 100000 );
```
Comment (rodydavis) [inline plugin/tests/e2e/config/bootstrap.js:208]: whats the reason for the increase 10x?

File: theme/assets/css/src/components/top-app-bar.css [hunk 10]
```diff
@@ -145,4 +145,13 @@
 	& .mdc-drawer__close {
 		display: none;
 	}
+
+	& .dark-mode__button {
```
Comment (rodydavis) [inline theme/assets/css/src/components/top-app-bar.css:149]: Is there a fallback?

**Review by rodydavis — CHANGES_REQUESTED** [review]
Just a few changes needed, and curious about the nested css.

File: plugin/assets/css/src/admin.css (hunk 1 above)
Comment (emeaguiar) [reply plugin/assets/css/src/admin.css:96]: @rodydavis There's a post-css plugin installed that allows us to use sass-like selectors

File: plugin/assets/css/src/components/material.css (hunk 2 above)
Comment (emeaguiar) [reply plugin/assets/css/src/components/material.css:33]: PostCSS turns this into `p .material-icons` in the compiled file. The final version will never see the `@nest` keyword

File: plugin/assets/css/src/customize-controls.css (hunk 3 above)
Comment (emeaguiar) [reply plugin/assets/css/src/customize-controls.css:63]: Same, the compiled file will read `.customize-control-image-radio-wrap h3`

File: plugin/assets/css/src/wizard/content.css (hunk 6 above)
Comment (emeaguiar) [reply plugin/assets/css/src/wizard/content.css:45]: @rodydavis Yes the plugin turns it into `.material-wizard__content .mdc-button--offsite .mdc-button__icon` in the compiled file

File: plugin/tests/e2e/config/bootstrap.js (hunk 9 above)
Comment (emeaguiar) [reply plugin/tests/e2e/config/bootstrap.js:208]: @rodydavis Error on my part, my computer keeps timing out on tests, I'll revert this

File: plugin/assets/src/customizer/components/color-control/style.css (hunk 7 above)
Comment (emeaguiar) [reply plugin/assets/src/customizer/components/color-control/style.css:6]: @rodydavis This is turned into `.material-design-color__picker .components-base-control .components-base-control__field` in the compiled file

File: plugin/assets/src/customizer/components/style-settings-control/style.css (hunk 8 above)
Comment (emeaguiar) [reply plugin/assets/src/customizer/components/style-settings-control/style.css:19]: @rodydavis This turns into `.style-settings-group .style-settings-group__title` in the compiled file

File: theme/assets/css/src/components/top-app-bar.css (hunk 10 above)
Comment (emeaguiar) [reply theme/assets/css/src/components/top-app-bar.css:149]: @rodydavis This turns into `.top-app-bar .dark-mode__button` in the compiled file

**Review by rodydavis — APPROVED** [review]
LGTM

**Comment by rodydavis** [comment]
Need some CLAs approved

**Comment by emeaguiar** [comment]
@googlebot I consent.

**Comment by PatelUtkarsh** [comment]
@googlebot I consent.

**Comment by rodydavis** [comment]
@googlebot I consent.

**Comment by rodydavis** [comment]
LGTM

**Pull request merged**